### ISSUE TITLE
[PhpUnitBridge] Composer-cache is not used

### ISSUE THREAD

**Issue by alexgit2k** [symfony/symfony#37652]
**Description**
We are using docker images for running tests where the downloaded composer-packages are cached between
the docker images to reduce download time. However, the composer packages for PHP-Unit are not cached
and downloaded every time which increases the start-time over 30 seconds.

**How to reproduce**
```
mkdir test && cd test
export COMPOSER_CACHE_DIR=composer_cache/
composer require symfony/phpunit-bridge
vendor/bin/simple-phpunit --help

rm -rf vendor
composer require symfony/phpunit-bridge
vendor/bin/simple-phpunit --help
```

**Possible Solution**
Cache vendor/bin/.phpunit, but this is the final installation directory.

**Comment by nicolas-grekas** [comment]
You could run `vendor/bin/simple-phpunit install` in your Dockerfile while building the image.

**Comment by alexgit2k** [comment]
@nicolas-grekas that would require parts of the source-code (composer.json, ...) already hardcoded in the docker-image. And also the installed version is cached, this is the same workaround as caching vendor/bin/.phpunit

**Comment by stof** [comment]
could it be that the bridge does not properly forward env variables when calling composer, and so does not let composer see the custom `COMPOSER_CACHE_DIR` config ?

**Comment by alexgit2k** [comment]
@stof I just checked that. If `COMPOSER_CACHE_DIR` ist not set it would default to $HOME/.composer/cache
 (https://getcomposer.org/doc/03-cli.md#composer-cache-dir), then it would be able to load the packages from cache at the 2nd run of `vendor/bin/simple-phpunit --help`, but it is downloaded again on every run.

**Comment by stof** [comment]
Can you show the full output of running these commands ?

**Comment by alexgit2k** [comment]
@stof Here it is:
```
# docker run -it --rm php:7.4 /bin/bash
root@1d231e768a05:/# apt update >/dev/null 2>/dev/null && apt install -y git unzip >/dev/null 2>/dev/null
root@1d231e768a05:/# curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
All settings correct for using Composer
Downloading...

Composer (version 1.10.9) successfully installed to: /usr/local/bin/composer
Use it: php /usr/local/bin/composer

root@1d231e768a05:/# mkdir test && cd test
root@1d231e768a05:/test# export COMPOSER_CACHE_DIR=composer_cache/
root@1d231e768a05:/test# time composer require symfony/phpunit-bridge
Using version ^5.1 for symfony/phpunit-bridge
./composer.json has been created
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing symfony/phpunit-bridge (v5.1.3): Downloading (100%)
symfony/phpunit-bridge suggests installing symfony/error-handler (For tracking deprecated interfaces usages at runtime with DebugClassLoader)
Writing lock file
Generating autoload files
1 package you are using is looking for funding.
Use the `composer fund` command to find out more!

real    0m13.501s
user    0m2.762s
sys     0m0.210s
root@1d231e768a05:/test# time vendor/bin/simple-phpunit --version
No composer.json found in the current directory, showing available packages from packagist.org
Creating a "phpunit/phpunit" project at "./phpunit-8.3-0"
Installing phpunit/phpunit (8.3.5)
Plugins have been disabled.
  - Installing phpunit/phpunit (8.3.5): Downloading (100%)
Created project in /test/vendor/bin/.phpunit/phpunit-8.3-0
composer.json has been updated
Loading composer repositories with package information
Updating dependencies
Package operations: 23 installs, 0 updates, 0 removals
  - Installing phar-io/version (2.0.1): Downloading (100%)
  - Installing phar-io/manifest (1.0.3): Downloading (100%)
  - Installing phpunit/php-timer (2.1.2): Downloading (100%)
  - Installing sebastian/recursion-context (3.0.0): Downloading (100%)
  - Installing sebastian/exporter (3.1.2): Downloading (100%)
  - Installing sebastian/diff (3.0.2): Downloading (100%)
  - Installing sebastian/comparator (3.0.2): Downloading (100%)
  - Installing sebastian/object-reflector (1.1.1): Downloading (100%)
  - Installing sebastian/global-state (3.0.0): Downloading (100%)
  - Installing sebastian/object-enumerator (3.0.3): Downloading (100%)
  - Installing sebastian/resource-operations (2.0.1): Downloading (100%)
  - Installing sebastian/type (1.1.3): Downloading (100%)
  - Installing doctrine/instantiator (1.3.1): Downloading (100%)
  - Installing myclabs/deep-copy (1.10.1): Downloading (100%)
  - Installing theseer/tokenizer (1.2.0): Downloading (100%)
  - Installing sebastian/version (2.0.1): Downloading (100%)
  - Installing sebastian/environment (4.2.3): Downloading (100%)
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Downloading (100%)
  - Installing phpunit/php-text-template (1.2.1): Downloading (100%)
  - Installing phpunit/php-token-stream (3.1.1): Downloading (100%)
  - Installing phpunit/php-file-iterator (2.0.2): Downloading (100%)
  - Installing phpunit/php-code-coverage (7.0.10): Downloading (100%)
  - Installing symfony/phpunit-bridge (dev-master): Symlinking from /test/vendor/symfony/phpunit-bridge
Writing lock file
Generating optimized autoload files
3 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
PHPUnit 8.3.5 by Sebastian Bergmann and contributors.


real    0m43.558s
user    0m7.626s
sys     0m0.903s
root@1d231e768a05:/test# rm -rf vendor
root@1d231e768a05:/test# time composer require symfony/phpunit-bridge
Using version ^5.1 for symfony/phpunit-bridge
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing symfony/phpunit-bridge (v5.1.3): Loading from cache
symfony/phpunit-bridge suggests installing symfony/error-handler (For tracking deprecated interfaces usages at runtime with DebugClassLoader)
Generating autoload files
1 package you are using is looking for funding.
Use the `composer fund` command to find out more!

real    0m3.530s
user    0m2.337s
sys     0m0.156s
root@1d231e768a05:/test# time vendor/bin/simple-phpunit --version
No composer.json found in the current directory, showing available packages from packagist.org
Creating a "phpunit/phpunit" project at "./phpunit-8.3-0"
Installing phpunit/phpunit (8.3.5)
Plugins have been disabled.
  - Installing phpunit/phpunit (8.3.5): Downloading (100%)
Created project in /test/vendor/bin/.phpunit/phpunit-8.3-0
composer.json has been updated
Loading composer repositories with package information
Updating dependencies
Package operations: 23 installs, 0 updates, 0 removals
  - Installing phar-io/version (2.0.1): Downloading (100%)
  - Installing phar-io/manifest (1.0.3): Downloading (100%)
  - Installing phpunit/php-timer (2.1.2): Downloading (100%)
  - Installing sebastian/recursion-context (3.0.0): Downloading (100%)
  - Installing sebastian/exporter (3.1.2): Downloading (100%)
  - Installing sebastian/diff (3.0.2): Downloading (100%)
  - Installing sebastian/comparator (3.0.2): Downloading (100%)
  - Installing sebastian/object-reflector (1.1.1): Downloading (100%)
  - Installing sebastian/global-state (3.0.0): Downloading (100%)
  - Installing sebastian/object-enumerator (3.0.3): Downloading (100%)
  - Installing sebastian/resource-operations (2.0.1): Downloading (100%)
  - Installing sebastian/type (1.1.3): Downloading (100%)
  - Installing doctrine/instantiator (1.3.1): Downloading (100%)
  - Installing myclabs/deep-copy (1.10.1): Downloading (100%)
  - Installing theseer/tokenizer (1.2.0): Downloading (100%)
  - Installing sebastian/version (2.0.1): Downloading (100%)
  - Installing sebastian/environment (4.2.3): Downloading (100%)
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Downloading (100%)
  - Installing phpunit/php-text-template (1.2.1): Downloading (100%)
  - Installing phpunit/php-token-stream (3.1.1): Downloading (100%)
  - Installing phpunit/php-file-iterator (2.0.2): Downloading (100%)
  - Installing phpunit/php-code-coverage (7.0.10): Downloading (100%)
  - Installing symfony/phpunit-bridge (dev-master): Symlinking from /test/vendor/symfony/phpunit-bridge
Writing lock file
Generating optimized autoload files
3 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
PHPUnit 8.3.5 by Sebastian Bergmann and contributors.


real    0m34.214s
user    0m8.050s
sys     0m0.847s
root@1d231e768a05:/test#
```

At the second `composer require symfony/phpunit-bridge` ithe packages are loaded from cache, but at the second `vendor/bin/simple-phpunit --version` the packages are downloaded again.

**Comment by svnldwg** [comment]
@alexgit2k did you find a solution? I'm dealing with a similar problem right now.

**Comment by jderusse** [comment]
Should be fixed by #39909.
Workaround is to use an absolute path. `COMPOSER_CACHE_DIR="$PWD/composer_cache"`
